### ISSUE TITLE
Adding a slice too many times results in jit error

### ISSUE THREAD

**Issue by awni** [arrayfire/arrayfire#2281]
The following code breaks (only tested on GPU). Any ideas?

```
#include <arrayfire.h>

int main() {
  af::info();
  auto x = af::randn(1, 2);
  auto y = af::randn(2, 2);

  for (int i = 0; i < 100; i++) {
    x += y(0, af::span);
  }

  return 0;
}
```

The output of this function is:

```
ArrayFire v3.6.0 (CUDA, 64-bit Linux, build default)
Platform: CUDA Toolkit 9.1, Driver: 396.26
[0] Tesla M40, 11449 MB, CUDA Compute 5.2
terminate called after throwing an instance of 'af::exception'
  what():  ArrayFire Exception (Internal error:998):
In function std::vector<char> cuda::compileToPTX(const char*, std::__cxx11::string)
In file src/backend/cuda/jit.cpp:291
NVRTC Error(6): NVRTC_ERROR_COMPILATION


In function af::array& af::array::operator+=(const af::array&)
In file src/api/cpp/array.cpp:814
Aborted (core dumped)
```

Some observations:
1. Fails after the 99th iteration
2. 1D (instead of 2D) arrays also works
3. Not using operator() also works.
4. The same but with swapped dimensions works.

**Comment by awni** [comment]
Hi Mark,

Thanks for following up on this! I no longer see this issue with v3.6.1 so I will close the issue.

Best,
Awni

**Comment by mark-poscablo** [comment]
Thanks @awni! Glad to hear :)

**Comment by WilliamTambellini** [comment]
was probably the same root cause as INVALID_IMAGE : 
https://github.com/arrayfire/arrayfire/issues/2527
fixed in 3.6.4
